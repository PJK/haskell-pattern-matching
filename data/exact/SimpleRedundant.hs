g :: Bool -> Int
g True = 1
g True = 2
g False = 3
