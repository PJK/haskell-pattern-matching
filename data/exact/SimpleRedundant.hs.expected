{
    "tag": "AnalysisSuccess",
    "contents": [
        [
            [
                "g",
                {
                    "tag": "Redundant",
                    "contents": [
                        {
                            "tag": "ConstructorPattern",
                            "contents": [
                                "True",
                                []
                            ]
                        }
                    ]
                }
            ]
        ],
        [
            [
                "g",
                [
                    [
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            }
                        ],
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            }
                        ]
                    ]
                ]
            ]
        ]
    ]
}
